### PR TITLE
Migrate TravisCI to GitHub Actions & Update Tested Ruby Versions

### PR DESCRIPTION
**Pull request by mkasberg**
This repo hasn't received much attention recently. As such, the TravisCI
yaml file was referencing
[EOL](https://www.ruby-lang.org/en/downloads/branches/) versions of
ruby. Moreover, TravisCI itself isn't generally used for open source
anymore. There are heavy restrictions on build minutes, as noted
[here](https://github.com/jekyll/jekyll/pull/8492) in the core Jekyll
project.

This PR does the following:

 * Removes `.travis.yml`. We won't run jobs on TravisCI anymore.
 * Replaces it with `.github/workflows/ci.yml`. We'll start running CI
   on GitHub Actions.
 * Updates tested Ruby versions to 2.7, 3.0, 3.1, and JRuby 9.3.4. This
  matches the ruby versions currently supported/tested by [Jekyll core](https://github.com/jekyll/jekyll/blob/796ae15c31147d1980662744ef0f19a15a27cdee/.github/workflows/ci.yml#L20-L28).
 * Adds `matrix` to the gemspec. In Ruby 3.1+, `matrix` is [no longer a
   default gem](https://www.ruby-lang.org/en/news/2021/12/25/ruby-3-1-0-released/) so we need to
   declare a dependency on it.

This makes progress toward https://github.com/jekyll/classifier-reborn/issues/192. I plan to work toward supporting Ruby 3
with fast SVD support and Numo in classifier-reborn, but the first step
is getting CI working with the existing code.

### CODE REVIEW

**Comment by mkasberg** [comment]
For reference/comparison, here is the last CI run on TravisCI against Ruby versions 2.4, 2.6, and JRuby 9.2.5. https://travis-ci.org/github/jekyll/classifier-reborn/builds/727612969

**Comment by mkasberg** [comment]
And here's the GitHub Actions run on my own fork of this repo: https://github.com/mkasberg/classifier-reborn/actions/runs/2282758186

**Review by mattr- — COMMENTED** [review]
Would you mind explaining why `matrix` was added to the gemspec?

**Comment by mkasberg** [comment]
Sorry, it got buried in my second commit message and I neglected to pull it up to the PR description.

> One of the [changes](https://www.ruby-lang.org/en/news/2021/12/25/ruby-3-1-0-released/)
in Ruby 3.1 is that `matrix` is no longer a default gem. As such, we
need to add it as a dependency to our gemspec for compatibility with
Ruby 3.1.

**Comment by mattr-** [comment]
No worries. I was on mobile and didn't think to check the commits.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+on:
+  push:
+    branches:
+      - master
+      - "*-stable"
+  pull_request:
+    branches:
+      - master
+      - "*-stable"
```
Comment (mattr-) [inline .github/workflows/ci.yml:11]: Would you mind adding `main` to the list of branches here? While there aren't any plans yet to wholesale move repos away from using `master` as the default branch, adding `main` here will make that task easier.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      - name: "Set up ${{ matrix.label }}"
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby_version }}
+          bundler-cache: true
+      - name: Run Minitest based tests
+        run: bash script/test
```
Comment (mattr-) [inline .github/workflows/ci.yml:45]: If you'd like, you can remove `bash` here
```suggestion
        run: script/test
```

**Review by mattr- — COMMENTED** [review]
This is looking good. I think the only thing I'd like to have addressed before merging this is the addition of `main` to the branch triggers. Everything else is optional.

Thank you for taking the time to contribute! ❤️

**Review by mattr- — APPROVED** [review]

**Comment by mattr-** [comment]
@jekyllbot: merge +dev

**Comment by mattr-** [comment]
Thank you! 🎉

**Pull request merged**